### PR TITLE
Ttlcache

### PR DESCRIPTION
**Pull request by mvkski**
PR for https://github.com/mozilla/bugbug/issues/1342

I added unit tests and integration tests for the proposed cache, but haven't yet been able to get the http service running locally on my machine so have yet to verify end to end. I'll be asking questions on how to do so in matrix, since that seems like the most appropriate channel for the topic.

### CODE REVIEW

**Comment by mvkski** [comment]
The PR appears to be failing due to linting. Is there any way to run the linting rules locally? I'll be fixing these later, as it is very late here in the US

File: http_service/bugbug_http/utils.py [hunk 1]
```diff
@@ -30,3 +33,38 @@ def get_hgmo_stack(branch: str, revision: str) -> list:
...
+    self.items_last_touched[key] = datetime.datetime.now()
+    return item
+
+  def __contains__(self, key):
+    return key in self.items
+
+  def purge_expired_entries(self):
+    purge_entries_before = datetime.datetime.now() - self.ttl
+    for (key, time_last_touched) in list(self.items_last_touched.items()):
+      print(time_last_touched)
```
Comment (mvkski) [inline http_service/bugbug_http/utils.py]: Remove this! This was left in for testing, and I missed it before submitting PR

**Comment by marco-c** [comment]
> The PR appears to be failing due to linting. Is there any way to run the linting rules locally? I'll be fixing these later, as it is very late here in the US

You can use pre-commit, that's the easiest way.
Just install the hooks with `pre-commit install` and then whenever you commit the linting will run automatically.

To fix the current issues, since you have already committed, you can use `pre-commit run --all-files`.

**Comment by mvkski** [comment]
I have reposted this pr [here](https://github.com/mozilla/bugbug/pull/1386) using my other github account max-reborn, which I would prefer to use for non professional projects at this time

**Pull request closed without merging**